### PR TITLE
Mask IP Addresses to subnets

### PR DESCRIPTION
**Pull request by stefanbenten**
What: This PR updates the last_ip column to contain only the /24 network the IP of the host is residing in. That ensures we are not vulnerable to someone running 200 nodes on different IPs within the same subnet.

Why: Improves data distribution across entire IP Ranges.

Please describe the tests:
 - Test 1: Done, ensures that the IPv4 and IPv6 Masking works as expected.
 - Test 2: Not needed.
 
Please describe the performance impact: Slightly more conversion, but improved grouping performance for the node selection

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: pkg/overlay/cache.go [hunk 1]
```diff
@@ -408,3 +408,21 @@ func getIP(ctx context.Context, target string) (_ string, err error) {
...
+		return "", err
+	}
+
+	addr := net.ParseIP(ip)
+	if addr == nil {
+		return "", errors.New("invalid ip")
+	}
+
+	//Filter all IP Addresses into /24 Subnet's
+	mask := net.CIDRMask(24, 32)
```
Comment (thepaul) [inline pkg/overlay/cache.go]: We should check first if `len(addr) == 4`. If `len(addr) == 16` instead, we should not apply a mask (as discussed, there doesn’t seem to be any very reasonable way to guess the appropriate subnet size with ipv6 addresses, so we will punt on masking those for now)

Comment (stefanbenten) [reply pkg/overlay/cache.go]: Ill add the check and the migration 👍

Comment (thepaul) [reply pkg/overlay/cache.go]: For the migration, `UPDATE nodes SET last_ip = host(network(set_masklen(last_ip::INET, 24))) WHERE last_ip <> '' AND family(last_ip::INET) = 4;` ought to make the necessary changes

Comment (stefanbenten) [reply pkg/overlay/cache.go]: I included a grouping per /64 subnet. I know that is not the best way, but better than allowing it entirely there..

**Comment by littleskunk** [comment]
@thepaul we need one more approval.

File: pkg/overlay/cache.go [hunk 2]
```diff
@@ -399,7 +399,7 @@ func (cache *Cache) GetMissingPieces(ctx context.Context, pieces []*pb.RemotePie
 	return missingPieces, nil
 }
 
-func getIP(ctx context.Context, target string) (_ string, err error) {
+func getIP(ctx context.Context, target string) (ip string, err error) {
```
Comment (jtolds) [inline pkg/overlay/cache.go]: can we have this be a net.IPAddr return value? that way GetNetwork doesn't need to call net.ParseIP

**Comment by stefanbenten** [comment]
> someday we should rename the last_ip column to last_ip_net or something

Renamed it to last_net 👍

File: pkg/overlay/cache.go [hunk 3]
```diff
@@ -399,15 +399,39 @@ func (cache *Cache) GetMissingPieces(ctx context.Context, pieces []*pb.RemotePie
 	return missingPieces, nil
 }
 
-func getIP(ctx context.Context, target string) (_ string, err error) {
+func getIP(ctx context.Context, target string) (ip net.IPAddr, err error) {
```
Comment (ifraixedes) [inline pkg/overlay/cache.go]: the `ip` return parameter is declared and never used, it was not needed to be changed from `_`

**Pull request merged**